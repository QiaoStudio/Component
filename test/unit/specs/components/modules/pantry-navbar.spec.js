import {
  shallowMount
} from 'dp-test'
import ComNavbar from 'components/modules/navbar/navbar.vue'
import ComMegamenu from 'elements/megamenu/megamenu.vue'
import ComButton from 'elements/button/button.vue'
import sinon from 'sinon'
import _ from 'lodash'

const brand = {
  'country': '',
  'type': '',
  'fallbackLogo': 'https://picsum.photos/121/36',
  'fallbackTagline': 'https://picsum.photos/121/11'
}

const navbarItems = [
  {
    'label': 'Insurance',
    'active': false,
    'url': '',
    'megamenu': [
      {
        'title': 'Compare insurance',
        'type': 'link',
        'layout': 'two-column',
        'items': [
          {
            'iconType': 'product',
            'iconCategory': 'travel-insurance',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Travel Insurance',
            'url': '/sg/travel-insurance',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'car-insurance',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Car Insurance',
            'url': '/sg/car-insurance',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'health-insurance',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Driver & Passenger\' Personal Accident',
            'url': '/sg/personal-accident',
            'target': ''
          }
        ]
      },
      {
        'title': 'Promotion',
        'type': 'article',
        'layout': '',
        'items': [
          {
            'imgUrl': 'https://picsum.photos/120',
            'url': '/sg/promotion',
            'title': 'Learn a bunch of stuff here',
            'description': 'Don\'t miss the exclusive campaign because you don\'t want to',
            'target': '_self'
          }
        ]
      }
    ]
  },
  {
    'label': 'Credit cards',
    'url': 'https://www.xxxxxx.com/sg/credit-card',
    'active': false,
    'megamenu': [
      {
        'title': 'Compare credit cards',
        'type': 'link',
        'layout': 'two-column',
        'items': [
          {
            'iconType': 'brand',
            'iconCategory': 'promotion',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Promotion',
            'url': '/sg/promotion',
            'target': ''
          },
          {
            'iconType': 'brand',
            'iconCategory': 'reward',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Reward',
            'url': '/sg/reward',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'ewallet',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Ewallet',
            'url': '/sg/ewallet',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'robofinance',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Robofinance',
            'url': '/sg/robofinance',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'stocks',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Stocks',
            'url': '/sg/stocks',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'bonds',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Bonds',
            'url': '/sg/bonds',
            'target': ''
          }
        ]
      },
      {
        'title': 'GoBear Credit cards',
        'type': 'article',
        'layout': '',
        'items': [
          {
            'imgUrl': 'https://picsum.photos/120',
            'url': '/sg/credit-cards',
            'title': 'New Credit cards',
            'description': 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
            'target': '_self'
          }
        ]
      }
    ]
  },
  {
    'label': 'Loans',
    'active': false,
    'url': '',
    'megamenu': [
      {
        'title': 'Compare loans',
        'type': 'link',
        'layout': '',
        'items': [
          {
            'iconType': 'product',
            'iconCategory': 'personal-loan',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Personal Loan',
            'url': '/sg/personal-loan',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'home-loan',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Home Loan',
            'url': '/sg/home-loan',
            'target': ''
          }
        ]
      },
      {
        'title': 'GoBear TV',
        'type': 'article',
        'layout': '',
        'items': [
          {
            'imgUrl': 'https://picsum.photos/120',
            'url': '/sg/promotion',
            'title': 'EP 22: Top 5 personal loan myths',
            'description': 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
            'target': '_self'
          }
        ]
      }
    ]
  },
  {
    'label': 'Banking',
    'active': false,
    'url': '',
    'megamenu': [
      {
        'title': 'Compare banking products',
        'type': 'link',
        'layout': 'two-column',
        'items': [
          {
            'iconType': 'product',
            'iconCategory': 'savings-account',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Saving account',
            'url': '/sg/savings-account',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'fixed-deposit',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Fixed deposit',
            'url': '/sg/fixed-deposit',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'ewallet',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Ewallet',
            'url': '/sg/ewallet',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'robofinance',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Robofinance',
            'url': '/sg/robofinance',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'stocks',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Stocks',
            'url': '/sg/stocks',
            'target': ''
          },
          {
            'iconType': 'product',
            'iconCategory': 'bonds',
            'iconDisplay': 'dp-icon--sm-only',
            'label': 'Bonds',
            'url': '/sg/bonds',
            'target': ''
          }
        ]
      },
      {
        'title': 'Blog',
        'type': 'article',
        'layout': '',
        'items': [
          {
            'imgUrl': 'https://picsum.photos/120',
            'url': '/sg/promotion',
            'title': 'Learn a bunch of stuff here',
            'description': 'Don\'t miss the exclusive campaign because you don\'t want to',
            'target': '_self'
          }
        ]
      }
    ]
  },
  {
    'label': 'Blog',
    'url': 'https://www.xxxxxx.com/sg/blog',
    'active': false,
    'megamenu': []
  }
]

function createWrapper(options) {
  let shallowOption = _.merge({
    propsData: {
      brand: brand,
      items: navbarItems
    },
    stubs: {
      'dp-button': ComButton,
      'dp-megamenu': ComMegamenu
    }
  }, options || {})
  return shallowMount(ComNavbar, shallowOption)
}

function createMockDiv(width, height) {
  const div = document.createElement('div')
  Object.assign(div.style, {
    width: width + 'px',
    height: height + 'px'
  })
  div.getBoundingClientRect = () => ({
    width,
    height,
    top: 0,
    left: 0,
    right: width,
    bottom: height
  })
  return div
}

describe('modules/Com Navbar', () => {
  describe('cases without Timers', () => {
    let wrapper
    let sanbox

    beforeEach(() => {
      sanbox = sinon.createSandbox()
      wrapper = createWrapper()
    })

    afterEach(() => {
      wrapper.destroy()
      sanbox.restore()
    })

    it('when navbar is scrolled, navbar scroll should be triggered', () => {
      wrapper = createWrapper()
      wrapper.vm.handleScroll()
      expect(wrapper.vm.isScrolled).toBe(false)
      sanbox.stub(window, 'scrollY').value(4)
      wrapper.vm.handleScroll()
      expect(wrapper.vm.isScrolled).toBe(true)
    })

    it('when navbar is active and not fixed, body should be overflowX hidden', () => {
      wrapper = createWrapper()
      wrapper.setData({
        isActive: true,
        isFixed: false
      })
      wrapper.vm.toggleOverflow()
      expect(document.body.style.overflowX).toBe('hidden')
      wrapper.setData({
        isActive: true,
        isFixed: true
      })
      wrapper.vm.toggleOverflow()
      expect(document.body.style.overflowX).toBe('')
    })

    it('when navbar content is positioned, navbar content should be positioned correctly', () => {
      let el = null
      wrapper = createWrapper()
      el = createMockDiv(1000, 283)
      wrapper.vm.subPosition(3, el, 1001)
      expect(wrapper.findAll('.dp-navbar__bg').at(0).attributes().style).toBe('transform: translate(1001px) scale(2.5, 1.415);')
      el = createMockDiv(300, 200)
      wrapper.vm.subPosition(3, el, 299)
      expect(wrapper.findAll('.dp-navbar__bg').at(0).attributes().style).toBe('transform: translate(299px) scale(0.75, 1);')
    })

    it('when navbar items is set, navbar items length should be correct', () => {
      wrapper = createWrapper()
      expect(wrapper.findAll('.dp-navbar__item').length).toBe(5)
    })

    it('when navbar transition started, navbar transition enter should be triggered', () => {
      wrapper = createWrapper()
      wrapper.vm.transEnter(wrapper.vm.$el.querySelector('.dp-navbar__sub'))
      expect(wrapper.findAll('.dp-navbar__bg').at(0).attributes().style).toBe('transform: translate(0px) scale(0, 0);')
      expect(wrapper.findAll('.dp-navbar__content').at(0).attributes().style).toBe('transform: translateX(0px);')
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('when navbar toggle is clicked, openMobileMenu should be emitted', () => {
      wrapper = createWrapper()
      wrapper.findAll('.dp-navbar__toggle').at(0).trigger('click')
      expect(wrapper.vm.$root.__emitted).toHaveProperty('openMobileMenu')
    })

    it('props items should be default []', () => {
      wrapper = shallowMount(ComNavbar, _.merge({
        propsData: {
          brand: brand
        }
      }))
      expect(wrapper.vm.menuItems).toEqual([])
    })

    it('when call showContent should change item correctly', () => {
      let event = {}
      wrapper = createWrapper({
        propsData: {
          items: [
            {
              'label': 'A',
              'url': 'https://www.xxxxxx.com/sg/a',
              'active': false,
              'megamenu': []
            },
            {
              'label': 'B',
              'url': 'https://www.xxxxxx.com/sg/b',
              'active': false,
              'megamenu': []
            }
          ]
        }
      })
      expect(wrapper.vm.menuItems[1].active).toBe(false)
      wrapper.vm.showContent(event, 1)
      expect(wrapper.vm.menuItems[1].active).toBe(true)
    })
  })
  describe('cases with Timers', () => {
    let wrapper
    let sanbox
    let clock

    beforeEach(() => {
      sanbox = sinon.createSandbox()
      wrapper = createWrapper()
      clock = sinon.useFakeTimers()
    })

    afterEach(() => {
      wrapper.destroy()
      sanbox.restore()
      clock.restore()
    })

    it('when navbar item content is hovered, navbar item content should be shown', () => {
      wrapper = createWrapper()
      wrapper.findAll('.dp-navbar__item').at(3).trigger('mouseenter')
      expect(wrapper.vm.menuItems[3].active).toBe(true)
      wrapper.findAll('.dp-navbar__item').at(3).trigger('mouseleave')
      clock.tick(300)
      expect(wrapper.vm.menuItems[3].active).toBe(false)
    })

    it('when navbar item sub is hovered, navbar item sub should be shown', () => {
      wrapper = createWrapper()
      wrapper.findAll('.dp-navbar__sub').at(0).trigger('mouseenter')
      expect(wrapper.vm.menuItems[0].active).toBe(true)
      expect(wrapper.vm.isActive).toBe(true)
      wrapper.findAll('.dp-navbar__sub').at(0).trigger('mouseleave')
      expect(wrapper.vm.isActive).toBe(false)
      clock.tick(300)
      expect(wrapper.vm.menuItems[0].active).toBe(false)
    })

    it('when showContent is called, navbar toggleOverflow should be triggered', () => {
      let toggleOverflowStub = sanbox.stub()
      let event = {}
      wrapper = createWrapper({
        propsData: {
          items: [
            {
              'label': 'A',
              'url': 'https://www.xxxxxx.com/sg/a',
              'active': false,
              'megamenu': []
            },
            {
              'label': 'B',
              'url': 'https://www.xxxxxx.com/sg/b',
              'active': false,
              'megamenu': []
            }
          ]
        },
        methods: {
          toggleOverflow: toggleOverflowStub
        }
      })
      wrapper.vm.showContent(event, 1)
      clock.tick(300)
      expect(toggleOverflowStub.called).toBe(true)
      expect(toggleOverflowStub.calledOnce).toBe(true)
    })

    it('when hideContent is called, navbar toggleOverflow should be triggered', () => {
      let toggleOverflowStub = sanbox.stub()
      let event = {}
      wrapper = createWrapper({
        propsData: {
          items: [
            {
              'label': 'A',
              'url': 'https://www.xxxxxx.com/sg/a',
              'active': false,
              'megamenu': []
            },
            {
              'label': 'B',
              'url': 'https://www.xxxxxx.com/sg/b',
              'active': false,
              'megamenu': []
            }
          ]
        },
        methods: {
          toggleOverflow: toggleOverflowStub
        }
      })
      wrapper.vm.hideContent(event, 1)
      clock.tick(300)
      expect(toggleOverflowStub.called).toBe(true)
      expect(toggleOverflowStub.calledOnce).toBe(true)
    })

    it('when call hideContent and item.active is false, should not change data', () => {
      let event = {}
      wrapper = createWrapper({
        propsData: {
          items: [
            {
              'label': 'A',
              'url': 'https://www.xxxxxx.com/sg/a',
              'active': false,
              'megamenu': []
            },
            {
              'label': 'B',
              'url': 'https://www.xxxxxx.com/sg/b',
              'active': false,
              'megamenu': []
            }
          ]
        },
        data() {
          return {
            noTransition: false,
            leftTimeout: null,
            showBackground: true
          }
        }
      })
      expect(wrapper.vm.menuItems[1].active).toBe(false)
      expect(wrapper.vm.leftTimeout).toBe(null)
      expect(wrapper.vm.showBackground).toBe(true)
      expect(wrapper.vm.noTransition).toBe(false)
      wrapper.vm.hideContent(event, 1)
      expect(wrapper.vm.leftTimeout).not.toBe(null)
      clock.tick(300)
      expect(wrapper.vm.menuItems[1].active).toBe(false)
      expect(wrapper.vm.showBackground).toBe(false)
      expect(wrapper.vm.noTransition).toBe(true)
    })
  })
})
