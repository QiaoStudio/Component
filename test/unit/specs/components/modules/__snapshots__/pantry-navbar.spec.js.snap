// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/Com Navbar cases without Timers when navbar transition started, navbar transition enter should be triggered 1`] = `
<header class="dp-navbar dp-navbar--fixed">
  <div class="dp-navbar__menu dp-navbar__no-transition">
    <div class="dp-navbar__brand">
      <dp-brand-stub country="" type="" fallbacklogo="https://picsum.photos/121/36" fallbacktagline="https://picsum.photos/121/11"></dp-brand-stub>
    </div>
    <div class="dp-navbar__items">
      <div class="dp-navbar__items--desktop">
        <div class="dp-navbar__bg-wrapper">
          <div class="dp-navbar__bg" style="transform: translate(0px) scale(0, 0);"></div>
        </div>
        <nav>
          <ul>
            <li class="dp-navbar__item">
              <a href="#"><span>
                    Insurance
                    <i aria-hidden="true" class="fa fa-caret-down dp-button__icon--right"></i></span></a>
            </li>
            <li class="dp-navbar__item">
              <a href="https://www.xxxxxx.com/sg/credit-card"><span>
                    Credit cards
                    <i aria-hidden="true" class="fa fa-caret-down dp-button__icon--right"></i></span></a>
            </li>
            <li class="dp-navbar__item">
              <a href="#"><span>
                    Loans
                    <i aria-hidden="true" class="fa fa-caret-down dp-button__icon--right"></i></span></a>
            </li>
            <li class="dp-navbar__item">
              <a href="#"><span>
                    Banking
                    <i aria-hidden="true" class="fa fa-caret-down dp-button__icon--right"></i></span></a>
            </li>
            <li>
              <a href="https://www.xxxxxx.com/sg/blog" class="dp-button dp-navbar__item dp-button--sm dp-button--text"><span>Blog</span>
                <link class="rippleJS">
              </a>
            </li>
            <!---->
          </ul>
        </nav>
      </div>
      <div class="dp-navbar__items--mobile">
        <a href="#" class="dp-navbar__toggle"><span>&nbsp;</span> <span>&nbsp;</span> <span>&nbsp;</span></a>
      </div>
    </div>
  </div>
  <div class="dp-navbar__content" style="transform: translateX(0px);"><span name="no-mode-translate-fade" tag="ul"><li data-sub-index="0" class="dp-navbar__sub" style="display: none;"><div class="dp-megamenu"><div><span>Compare insurance</span>
    <ul class="dp-megamenu__link--two-column">
      <li class="dp-megamenu__link">
        <a href="/sg/travel-insurance" target="_self"><span class="dp-icon dp-icon__product--travel-insurance dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Travel Insurance</span></a>
      </li>
      <li class="dp-megamenu__link">
        <a href="/sg/car-insurance" target="_self"><span class="dp-icon dp-icon__product--car-insurance dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Car Insurance</span></a>
      </li>
      <li class="dp-megamenu__link">
        <a href="/sg/personal-accident" target="_self"><span class="dp-icon dp-icon__product--health-insurance dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Driver &amp; Passenger' Personal Accident</span></a>
      </li>
    </ul>
  </div>
  <div><span>Promotion</span>
    <div>
      <div class="dp-megamenu__article">
        <a href="/sg/promotion" target="_self">
          <div class="dp-megamenu__article-img">
            <img src="https://picsum.photos/120" alt="Learn a bunch of stuff here">
          </div>
          <div><span class="dp-megamenu__article-title">Learn a bunch of stuff here</span>
            <p class="dp-megamenu__article-desc">Don't miss the exclusive campaign because you don't want to</p>
          </div>
        </a>
      </div>
    </div>
  </div>
  </div>
  </li>
  <li data-sub-index="1" class="dp-navbar__sub" style="display: none;">
    <div class="dp-megamenu">
      <div><span>Compare credit cards</span>
        <ul class="dp-megamenu__link--two-column">
          <li class="dp-megamenu__link">
            <a href="/sg/promotion" target="_self"><span class="dp-icon dp-icon__brand--promotion dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Promotion</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/reward" target="_self"><span class="dp-icon dp-icon__brand--reward dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Reward</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/ewallet" target="_self"><span class="dp-icon dp-icon__product--ewallet dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Ewallet</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/robofinance" target="_self"><span class="dp-icon dp-icon__product--robofinance dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Robofinance</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/stocks" target="_self"><span class="dp-icon dp-icon__product--stocks dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Stocks</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/bonds" target="_self"><span class="dp-icon dp-icon__product--bonds dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Bonds</span></a>
          </li>
        </ul>
      </div>
      <div><span>GoBear Credit cards</span>
        <div>
          <div class="dp-megamenu__article">
            <a href="/sg/credit-cards" target="_self">
              <div class="dp-megamenu__article-img">
                <img src="https://picsum.photos/120" alt="New Credit cards">
              </div>
              <div><span class="dp-megamenu__article-title">New Credit cards</span>
                <p class="dp-megamenu__article-desc">Lorem ipsum dolor sit amet, consectetur adipiscing elit</p>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li data-sub-index="2" class="dp-navbar__sub" style="display: none;">
    <div class="dp-megamenu">
      <div><span>Compare loans</span>
        <ul class="">
          <li class="dp-megamenu__link">
            <a href="/sg/personal-loan" target="_self"><span class="dp-icon dp-icon__product--personal-loan dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Personal Loan</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/home-loan" target="_self"><span class="dp-icon dp-icon__product--home-loan dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Home Loan</span></a>
          </li>
        </ul>
      </div>
      <div><span>GoBear TV</span>
        <div>
          <div class="dp-megamenu__article">
            <a href="/sg/promotion" target="_self">
              <div class="dp-megamenu__article-img">
                <img src="https://picsum.photos/120" alt="EP 22: Top 5 personal loan myths">
              </div>
              <div><span class="dp-megamenu__article-title">EP 22: Top 5 personal loan myths</span>
                <p class="dp-megamenu__article-desc">Lorem ipsum dolor sit amet, consectetur adipiscing elit</p>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li data-sub-index="3" class="dp-navbar__sub" style="display: none;">
    <div class="dp-megamenu">
      <div><span>Compare banking products</span>
        <ul class="dp-megamenu__link--two-column">
          <li class="dp-megamenu__link">
            <a href="/sg/savings-account" target="_self"><span class="dp-icon dp-icon__product--savings-account dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Saving account</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/fixed-deposit" target="_self"><span class="dp-icon dp-icon__product--fixed-deposit dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Fixed deposit</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/ewallet" target="_self"><span class="dp-icon dp-icon__product--ewallet dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Ewallet</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/robofinance" target="_self"><span class="dp-icon dp-icon__product--robofinance dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Robofinance</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/stocks" target="_self"><span class="dp-icon dp-icon__product--stocks dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Stocks</span></a>
          </li>
          <li class="dp-megamenu__link">
            <a href="/sg/bonds" target="_self"><span class="dp-icon dp-icon__product--bonds dp-icon--sm-only"></span> <span class="dp-megamenu__link-label">Bonds</span></a>
          </li>
        </ul>
      </div>
      <div><span>Blog</span>
        <div>
          <div class="dp-megamenu__article">
            <a href="/sg/promotion" target="_self">
              <div class="dp-megamenu__article-img">
                <img src="https://picsum.photos/120" alt="Learn a bunch of stuff here">
              </div>
              <div><span class="dp-megamenu__article-title">Learn a bunch of stuff here</span>
                <p class="dp-megamenu__article-desc">Don't miss the exclusive campaign because you don't want to</p>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li data-sub-index="4" class="dp-navbar__sub" style="display: none;">
    <div class="dp-megamenu"></div>
  </li>
  </span>
  </div>
</header>
`;
